File: api/spring-boot-api/src/main/java/com/spicytodo/model/TodoStats.java
package com.spicytodo.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TodoStats {
    private int total;
    private int active;
    private int completed;
    private double completionRate;
    private Map<String, Integer> priorityBreakdown;
    private int overdueCount;
    private int dueTodayCount;
    private int upcomingCount;
}

File: api/dotnet-api/Models/TodoStats.cs
namespace DotnetApi.Models
{
    public class TodoStats
    {
        public int Total { get; set; }
        public int Active { get; set; }
        public int Completed { get; set; }
        public double CompletionRate { get; set; }
        public Dictionary<string, int> PriorityBreakdown { get; set; } = new();
        public int OverdueCount { get; set; }
        public int DueTodayCount { get; set; }
        public int UpcomingCount { get; set; }
    }
}

